Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,714 @@
+  ⟨fun _ _ => ext fun _ => op_smul_eq_smul _ _⟩
+
+instance instSMulWithZero [Zero β] {R : Type*} [Zero R] [SMulWithZero R β]
+    [ContinuousConstSMul R β] : SMulWithZero R C_c(α, β) :=
+  Function.Injective.smulWithZero ⟨_, coe_zero⟩ DFunLike.coe_injective coe_smul
+
+instance instMulActionWithZero [Zero β] {R : Type*} [MonoidWithZero R] [MulActionWithZero R β]
+    [ContinuousConstSMul R β] : MulActionWithZero R C_c(α, β) :=
+  Function.Injective.mulActionWithZero ⟨_, coe_zero⟩ DFunLike.coe_injective coe_smul
+
+instance instModule [AddCommMonoid β] [ContinuousAdd β] {R : Type*} [Semiring R] [Module R β]
+    [ContinuousConstSMul R β] : Module R C_c(α, β) :=
+  Function.Injective.module R ⟨⟨_, coe_zero⟩, coe_add⟩ DFunLike.coe_injective coe_smul
+
+instance instNonUnitalNonAssocSemiring [NonUnitalNonAssocSemiring β] [TopologicalSemiring β] :
+    NonUnitalNonAssocSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance instNonUnitalSemiring [NonUnitalSemiring β] [TopologicalSemiring β] :
+    NonUnitalSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance instNonUnitalCommSemiring [NonUnitalCommSemiring β] [TopologicalSemiring β] :
+    NonUnitalCommSemiring C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommSemiring _ coe_zero coe_add coe_mul fun _ _ => rfl
+
+instance instNonUnitalNonAssocRing [NonUnitalNonAssocRing β] [TopologicalRing β] :
+    NonUnitalNonAssocRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalNonAssocRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instNonUnitalRing [NonUnitalRing β] [TopologicalRing β] : NonUnitalRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalRing _ coe_zero coe_add coe_mul coe_neg coe_sub (fun _ _ => rfl)
+    fun _ _ => rfl
+
+instance instNonUnitalCommRing [NonUnitalCommRing β] [TopologicalRing β] :
+    NonUnitalCommRing C_c(α, β) :=
+  DFunLike.coe_injective.nonUnitalCommRing _ coe_zero coe_add coe_mul coe_neg coe_sub
+    (fun _ _ => rfl) fun _ _ => rfl
+
+instance instIsScalarTower {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [IsScalarTower R β β] :
+    IsScalarTower R C_c(α, β) C_c(α, β) where
+  smul_assoc r f g := by
+    ext
+    simp only [smul_eq_mul, coe_mul, coe_smul, Pi.mul_apply, Pi.smul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_assoc]
+
+instance instSMulCommClass {R : Type*} [Semiring R] [NonUnitalNonAssocSemiring β]
+    [TopologicalSemiring β] [Module R β] [ContinuousConstSMul R β] [SMulCommClass R β β] :
+    SMulCommClass R C_c(α, β) C_c(α, β) where
+  smul_comm r f g := by
+    ext
+    simp only [smul_eq_mul, coe_smul, coe_mul, Pi.smul_apply, Pi.mul_apply]
+    rw [← smul_eq_mul, ← smul_eq_mul, smul_comm]
+
+end AlgebraicStructure
+
+section ZeroAtInfty
+
```

Comment:
You should exit the namespace `CompactlySupportedContinuousMap` for this section, as it has broader application range, and also for the other sections which are about `CompactlySupportedContinuousMapClass`.

Reply:
I move the section at the end out of `CompactlySupportedContinuousMap`.